pkg/apis/core/validation/validation.go

```diff
@@ -5283,11 +5283,20 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 // maxGMSACredentialSpecLength is the max length, in bytes, for the actual contents
 // of a GMSA cred spec. In general, those shouldn't be more than a few hundred bytes,
 // so we want to give plenty of room here while still providing an upper bound.
+// maxRunAsUserNameLength is the max character length for the username which will be
+// used to execute the given container's entrypoint. The username cannot be empty,
+// must contain at most 20 characters, cannot contain only spaces and periods (.),
+// and it cannot contain the following characters: " / \ [ ] : ; | = , + * ? < > @
```

Comment:
`\` will be allowed to handle domain-name prefix as discussed.

Reply:
if we're allowing domain-qualified names, I question whether the limits for local usernames should still be added. length especially seems problematic if the name can contain a domain qualification, and I don't know whether any of the special characters are allowed in domain names

Reply:
[This](https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and) appears to be the main guidance around domain names (although it does not explicitly state Server 2019). For DNS Domain name prefixes, here are the relevant bits:

```
Maximum name length
255 characters.

The maximum length of the host name and of the fully qualified domain name (FQDN) is 63 bytes per label and 255 characters per FQDN.
```

Based on the above, `maxRunAsUserNameLength` may be capped to 21 (name) + 255 (FQDN) = 276.

Regarding special characters, it states:
```
DNS domain names cannot contain the following characters:
comma (,)
tilde (~)
colon (:)
exclamation point (!)
at sign (@)
number sign (#)
dollar sign ($)
percent (%)
caret (^)
ampersand (&)
apostrophe (')
period (.)
parentheses (())
braces ({})
underscore (_)
white space (blank)
```

So the common items/intersection of the above set for DNS domain names and the set of characters not allowed for usernames is: `:  , @`

Reply:
Done,

Although, I opted for a few things differently in order to make things more clear on error.